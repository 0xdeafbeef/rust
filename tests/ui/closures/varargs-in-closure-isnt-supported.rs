// var-args are not supported in closures, ensure we don't misdirect people (#146489)
#![feature(c_variadic)]

unsafe extern "C" fn thats_not_a_pattern(mut ap: ...) -> u32 {
    let mut lol = |...| (); //~ ERROR: unexpected `...`
    unsafe { ap.arg::<u32>() } //~^ NOTE: C-variadic type `...` is not allowed here
    //~| NOTE: not a valid pattern
}

fn main() {}
