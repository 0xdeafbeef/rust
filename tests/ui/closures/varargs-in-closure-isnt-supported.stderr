error: unexpected `...`
  --> $DIR/varargs-in-closure-isnt-supported.rs:5:20
   |
LL |     let mut lol = |...| ();
   |                    ^^^ not a valid pattern
   |
   = note: C-variadic type `...` is not allowed here

error[E0282]: type annotations needed
  --> $DIR/varargs-in-closure-isnt-supported.rs:5:20
   |
LL |     let mut lol = |...| ();
   |                    ^^^
   |
help: consider giving this closure parameter an explicit type
   |
LL |     let mut lol = |...: /* Type */| ();
   |                       ++++++++++++

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0282`.
